### PR TITLE
Scrapper, Kristina Chekmareva - 21FPL1

### PR DESCRIPTION
**Pull request by kschekmareva**

### CODE REVIEW

File: lab_5_scrapper/scrapper.py [hunk 1]
```diff
@@ -13,62 +52,92 @@ def __init__(self, path_to_config: Path) -> None:
...
         """
-        pass
+        self.path_to_config = path_to_config
+        self.config_dto = self._extract_config_content()
+        self._validate_config_content()
+        self._get_seed_urls = self.get_seed_urls()
+        self._num_articles = self.get_num_articles()
+        self._get_headers = self.get_headers()
+        self._get_encoding = self.get_encoding()
+        self._get_timeout = self.get_timeout()
```
Comment (dmitry-uraev) [inline lab_5_scrapper/scrapper.py]: class fields naming style should be different - why do you name class attribute using verb prefix `_get`?

Comment (dmitry-uraev) [reply lab_5_scrapper/scrapper.py]: it is not method

File: lab_5_scrapper/scrapper.py [hunk 2]
```diff
@@ -13,62 +52,92 @@ def __init__(self, path_to_config: Path) -> None:
...
+
+        if not isinstance(self.config_dto.verify_certificate, bool):
+            raise IncorrectVerifyError
 
     def get_seed_urls(self) -> list[str]:
         """
         Retrieve seed urls
         """
-        pass
+        return self.config_dto.seed_urls
```
Comment (dmitry-uraev) [inline lab_5_scrapper/scrapper.py]: If you get all information inside you getting methods using `self.config_dto.<something>`, so why do you define class attributes e.g. self._num_articles? How do you use  and get them, especially when these fields are protected (you use underscore in name)?

Try to reimplement this logic in another way - your getting methods e.g. `get_num_articles` should return `self._num_article` field value, as it is protected and there is no other way to get it.

**Review by dmitry-uraev — CHANGES_REQUESTED** [review]
Good start. Please, continue implementing lab 5.

File: lab_5_scrapper/scrapper.py [hunk 3]
```diff
@@ -90,25 +195,48 @@ def __init__(self, config: Config) -> None:
         """
         Initializes an instance of the Crawler class
         """
-        pass
+        self._seed_urls = config.get_seed_urls()
+        self.config = config
+        self.urls = []
 
-    def _extract_url(self, article_bs: BeautifulSoup) -> str:
+    @staticmethod
```
Comment (dmitry-uraev) [inline lab_5_scrapper/scrapper.py:202]: why do you use this decorator?

Comment (kschekmareva) [reply lab_5_scrapper/scrapper.py:202]: This method works only with the parameters passed to it. It does not use data from the class instance in any way.

**Review by kschekmareva — COMMENTED** [review]
/

File: lab_5_scrapper/scrapper.py (hunk 2 above)
Comment (kschekmareva) [reply lab_5_scrapper/scrapper.py]: fbvddbd

**Comment by vitazaytseva** [comment]
Well done! You can proceed to work on Lab 6)

**Pull request closed without merging**